apps/saber/Assets/Scripts/Interactions/PlayerInteract.cs

```diff
@@ -0,0 +1,63 @@
+using UnityEngine;
+
+public class PlayerInteract : MonoBehaviour
+{
+    [Tooltip("The Transform representing the player's position.")]
+    [SerializeField] Transform playerPos;
+
+    [Tooltip("Maximum distance for object interaction")]
+    [SerializeField] float interactionDistance = 3f;
+
+    [Tooltip("Layer(s) for interactable objects")]
+    [SerializeField] LayerMask interactableLayer;
+
+    [Tooltip("Key for interaction")]
+    [SerializeField] KeyCode interactionKey = KeyCode.E;
+
+    [SerializeField] bool onDebug = true;
+
+    private Ray ray; // Represents the interaction ray.
+    private RaycastHit raycastHit; // Stores information about what the ray hits.
+    void Update()
+    {
+        // Generate the interaction ray and display it for debugging.
+        GenerateRay();
+        DisplayRayOnDebug(onDebug);
+    }
+
+    void GenerateRay()
+    {
+        // Create a ray from the player's position in the direction they are facing.
+        ray = new Ray(playerPos.transform.position, playerPos.transform.forward);
```

Comment:
i think the ray should be shot from the camera (you can use the ScreenPointToRay function) so the player can more easily interact by looking at an object and not turning the character towards it